test/unit/sqs/sqs-consumer-spec.js

```diff
@@ -244,10 +248,156 @@ describe('SqsConsumer', () => {
+      consumer = new SqsConsumer({sqs: sqs}, msgBody => Promise.resolve());
+    });
+
     it('should stop polling', () => {
       consumer._running=true;
       consumer.stop();
       consumer._running.should.be.false;
     });
   });
-});
\ No newline at end of file
+
+  describe('_scheduledConsuming', () => {
+    let config = {
+          defaults: {
+            consumer: {
+              scheduler: {
+                scheduled: true,
+                start: '12:00:00',
+                duration: 2 * 60 * 60,
+                maxVisibilityTimeout: 10
+              }
+            }
+          }
+        },
+        messages = {
+          Messages: [{Body: "{}",ReceiptHandle: 'handle1'}]
+        };
+
+    beforeEach(() => {
+      config = {
+        defaults: {
+          consumer: {
+            scheduler: {
+              scheduled: true,
+              start: '12:00:00',
+              duration: 2 * 60 * 60,
+              maxVisibilityTimeout: 10
+            }
+          }
+        }
+      };
+    });
+
+    context('when consumer config is not present', () => {
+      before(() => {
+        delete config.defaults.consumer.scheduler;
+        consumer = new SqsConsumer({sqs: sqs}, msgBody => Promise.resolve(), {sqs: config});
+      });
+
+      it('resolves the original data', () => {
+        sqs.receiveMessage.returns({ promise: () => Promise.resolve(messages)});
+        return sqs.receiveMessage()
+          .promise()
+          .then(data => {
+            return consumer._scheduledConsuming(data);
+          })
+          .then(data => {
+            data.should.deep.equal(messages);
+          });
+      });
+    });
+
+    context('when consumer is not scheduled', () => {
+      before(() => {
+        config.defaults.consumer.scheduler.scheduled = false;
+        consumer = new SqsConsumer({sqs: sqs}, msgBody => Promise.resolve(), {sqs: config});
+      });
+
+      it('resolves the original data', () => {
+        sqs.receiveMessage.returns({ promise: () => Promise.resolve(messages)});
+        return sqs.receiveMessage()
+          .promise()
+          .then(data => {
+            return consumer._scheduledConsuming(data);
+          })
+          .then(data => {
+            data.should.deep.equal(messages);
+          });
+      });
+    });
+
+    context('when polling messages before the scheduled processing window', () => {
+      let clock;
+
+      before(() => {
+        consumer = new SqsConsumer({sqs: sqs}, msgBody => Promise.resolve(), {sqs: config});
+        clock = sinon.useFakeTimers(new Date().setHours(9, 0, 0));
+      });
+
+      after(() => {
+        clock.restore();
+      });
+
+      it('resolves an empty array', () => {
+        sqs.receiveMessage.returns({ promise: () => Promise.resolve(messages)});
+        return sqs.receiveMessage()
+          .promise()
+          .then(data => consumer._scheduledConsuming(data))
+          .then(data => {
+            data.should.deep.equal({});
```

Comment:
Suggestion: Would be great if we can also assert that message visibility is set to to required duration